packages/preferences/src/browser/workspace-preference-provider.ts

```diff
@@ -25,4 +28,12 @@ export class WorkspacePreferenceProvider extends AbstractResourcePreferenceProvi
         return new Promise<URI>(() => { });
     }
 
+    setPreference(key: string, value: string): Promise<void> {
+        return this.storageService.setData(key, value);
```

Comment:
Why is that? It should write to `.theia/setting.json` file in the workspace not in the local storage.

Reply:
Do you mean that I need to implement `WorkspaceStorageResource` similar to `UserStorageResource` ?

Reply:
Could you elaborate why `StorageService` should be used here? I think the default implementation in `AbstractResourcePreferenceProvider` should work. It saves changes to `.theia/setting.json` file. `StorageService` is an API to the session data.

Reply:
>  I think the default implementation in `AbstractResourcePreferenceProvider` should work

This method uses `ResourceProvider` to work with `setiings.json` file, and this provider is implemented by `UserStorageResource`. This provider is provided only by `UserPreferenceProvider`.